tests/test_tpa_pipeline.py

```diff
@@ -155,6 +155,33 @@ def test_handle_enterprise_logistration_consent_not_required(self):
                 enterprise_customer=enterprise_customer,
                 user_id=self.user.id
             ).count() == 1
+            assert UserDataSharingConsentAudit.objects.filter(
+                user__user_id=self.user.id,
+                user__enterprise_customer=enterprise_customer,
+            ).count() == 0
+
+    def test_handle_enterprise_logistration_consent_externally_managed(self):
+        """
+        Test that when consent is externally managed, we create an EnterpriseCustomerUser and
+        UserDataSharingConsentAudit object, then return.
+        """
+        backend = mock.MagicMock(name=None)
+        with mock.patch('enterprise.tpa_pipeline.get_ec_for_running_pipeline') as fake_get_ec:
```

Comment:
You can make this as a decorator and pass it as an arg to help avoid getting too indented. Like so:

```python
@mock.patch('enterprise.tpa_pipeline.get_ec_for_running_pipeline')
def test_handle_enterprise_logistration_consent_externally_managed(self, fake_get_ec):
    pass
```